### PR TITLE
new_feature: add clear all notifications button.

### LINKED ISSUE #3328: feat: clear all button for notifications
**Is your feature request related to a problem? Please describe.**
If the user wants to delete all the notification there need to be an option for that.

**Describe the solution you'd like**
A "Clear all" button similar to "mark all as read" that destroys all notifications. Before destroying, it should ask for confirmation in `modal`.

**Are you working on this? (Yes/No)**
No

### PR DESCRIPTION
**Pull request by palashb01**
Fixes #3328 

#### Describe the changes you have made in this PR -

- Added a button to clear all notifications.

### Demo Video

https://user-images.githubusercontent.com/66828942/200446714-a0ede24a-6341-4580-8971-7de6c28e58bd.mp4

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

**Comment by palashb01** [comment]
Hey @vedant-jain03 , Please review the PR, Thanks.

File: app/views/users/noticed_notifications/index.html.erb [hunk 1]
```diff
@@ -11,11 +11,20 @@
...
         <%= link_to t("users.notifications.mark_as_read_button"), mark_all_as_read_url(current_user), class: "anchor-text" %>
       </div>
     <% end %>
+    </div>
+    <div>
+    <% if @notification.count.positive? %>
+      <div>
+        <%= link_to t("users.notifications.delete_all_notifications"), delete_all_notifications_url(current_user), class: "anchor-text" %>
+      </div>
+  <% end %>
```
Comment (vedant-jain03) [inline app/views/users/noticed_notifications/index.html.erb:26]:
```suggestion
      <% if @unread.count.positive? %>
         <%= link_to t("users.notifications.mark_as_read_button"), mark_all_as_read_url(current_user), class: "anchor-text" %>
      <% end %>
      <% if @notification.count.positive? %>
        <%= link_to "Clear All", delete_all_notifications_url(current_user), class: "anchor-text" %>
      <% end %>
```

- Also "Delete All Notifications" looks very big, make it "Clear All".
- The button should first call for confirmation in the modal. You can refer to how the modal appears each time before asking to remove the "member" from the group.
- The Button color should be `$primary-red`

**Review by vedant-jain03 — CHANGES_REQUESTED** [review]
Make the text "Clear All" translatable!
You need to add [here](https://github.com/CircuitVerse/CircuitVerse/blob/d27a74baa875200e80733823e6be0d226d4e0ee6/config/locales/views/users/en.yml#L93)

File: app/views/users/noticed_notifications/index.html.erb (hunk 1 above)
Comment (palashb01) [reply app/views/users/noticed_notifications/index.html.erb:26]: Got it, Thanks for reviewing the PR, I will make the requested changes.

**Comment by palashb01** [comment]
Closing this PR, since multiple changes are required and a draft PR is requested for further review.

**Pull request closed without merging**